drivers/gpio/gpio_stm32.c

```diff
@@ -221,6 +222,51 @@ int gpio_stm32_configure(uint32_t *base_addr, int pin, int conf, int altf)
 	return 0;
 }
 
+/**
+ * @brief GPIO port clock handling
+ */
+int gpio_stm32_clock_request(const struct device *dev, bool on)
+{
+	const struct gpio_stm32_config *cfg = dev->config;
+	int ret = 0;
+
+	if (dev == NULL) {
+		return -ENODEV;
+	}
+
+	/* enable clock for subsystem */
+	const struct device *clk = DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE);
+
```

Comment:
I think you have to check if the clock device is ready at this point.

Reply:
`device_is_ready` check is done directly in API (clock_control.h)

Reply:
@alexanderwachter is that fine, with this info ?

Reply:
Think so, yes